app/controllers/concerns/json_pagination.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+module JsonPagination
+  private
+
+  def add_pagination_links(json, pagy)
+    # add custom headers to the response
+    add_headers(pagy)
+
+    # Insert links at beginning
+    links = links(pagy)
+    if links.present?
+      json_d = json.clone
+      json.clear
+      json["links"] = links
+      json.merge! json_d
+    end
+  end
+
+  def pages(pagy)
+    paging = {}
+    paging[:first] = 1          unless pagy.page == 1
+    paging[:last] = pagy.pages  unless pagy.page == pagy.pages
+
+    paging[:prev] = pagy.prev   unless pagy.prev.nil?
+    paging[:next] = pagy.next   unless pagy.next.nil?
+    paging
+  end
+
+  def links(pagy)
+    links = {}
+    uri = request.env["PATH_INFO"]
+
+    # Build pagination links
+    pages(pagy).each do |key, value|
+      query_params = request.query_parameters.merge(page: value)
+      links[key] = "#{uri}?#{query_params.to_param}"
+    end
+    links
+  end
+
+  def add_headers(pagy)
+    headers['X-PER-PAGE']      = pagy.offset
+    headers["X-CURRENT-PAGE"]  = pagy.page
+    headers["X-TOTAL-PAGES"]   = pagy.pages
+    headers["X-TOTAL-RECORDS"] = pagy.count
```

Comment:
is this stolen from somewhere/a common pattern ?
would it be easier to have `pagination` in the reply so that users don't have to read headers ?
(check what github or zendesk or openapi do)

Reply:
It was just inspired by Github API and some gems.
Where GitHub adds everything including links in the header. I prefer most frequently changing items like `links` should be the body and some constant values like.. `total_Recods`, `total_page` ..etc should be in the header.